state-chain/custom-rpc/src/lib.rs

```diff
@@ -1085,6 +1129,25 @@ where
 		self.new_update_subscription(sink, move |api, hash| api.cf_pool_price(hash, from, to))
 	}
 
+	fn cf_subscribe_pool_price_v2(
+		&self,
+		sink: SubscriptionSink,
+		base_asset: RpcAsset,
+		quote_asset: RpcAsset,
+	) -> Result<(), SubscriptionEmptyError> {
+		let base = base_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
+		let quote = quote_asset.try_into().map_err(|_| SubscriptionEmptyError)?;
+		self.new_update_subscription(sink, move |api, hash| {
+			let sell_price = api.cf_pool_price(hash, quote, base)?;
```

Comment:
I didn't add a pool_price_v2 to the Runtime API because the V2 only does what V1 does twice, I thought it was unnecessary to create a separate function for it. So here, I would do the same as the cf_pool_price_v2 and construct the PoolPriceV2 struct here. 
Does this look OK? I think the duplicate code is minimal